src/main/java/org/elasticsearch/common/rounding/TimeZoneRounding.java

```diff
@@ -362,94 +196,48 @@ public byte id() {
 
         @Override
         public long roundKey(long utcMillis) {
-            long time = utcMillis + preTz.getOffset(utcMillis);
-            return Rounding.Interval.roundKey(time, interval);
-        }
-
-        @Override
-        public long valueForKey(long key) {
-            long time = Rounding.Interval.roundValue(key, interval);
-            // now, time is still in local, move it to UTC
-            time = time - preTz.getOffset(time);
-            // now apply post Tz
-            time = time + postTz.getOffset(time);
-            return time;
-        }
-
-        @Override
-        public long nextRoundingValue(long value) {
-            return value + interval;
-        }
-
-        @Override
-        public void readFrom(StreamInput in) throws IOException {
-            interval = in.readVLong();
-            preTz = DateTimeZone.forID(in.readString());
-            postTz = DateTimeZone.forID(in.readString());
-        }
-
-        @Override
-        public void writeTo(StreamOutput out) throws IOException {
-            out.writeVLong(interval);
-            out.writeString(preTz.getID());
-            out.writeString(postTz.getID());
-        }
-    }
-
-    static class DayIntervalTimeZoneRounding extends TimeZoneRounding {
-
-        final static byte ID = 6;
-
-        private long interval;
-        private DateTimeZone preTz;
-        private DateTimeZone postTz;
-
-        DayIntervalTimeZoneRounding() { // for serialization
-        }
-
-        DayIntervalTimeZoneRounding(long interval, DateTimeZone preTz, DateTimeZone postTz) {
-            this.interval = interval;
-            this.preTz = preTz;
-            this.postTz = postTz;
-        }
-
-        @Override
-        public byte id() {
-            return ID;
-        }
-
-        @Override
-        public long roundKey(long utcMillis) {
-            long time = utcMillis + preTz.getOffset(utcMillis);
-            return Rounding.Interval.roundKey(time, interval);
+            long timeLocal = utcMillis;
+            if (!isUtc) {
+                timeLocal = timeZone.convertUTCToLocal(utcMillis);
+            }
+            return Rounding.Interval.roundKey(timeLocal, interval);
         }
 
         @Override
         public long valueForKey(long key) {
-            long time = Rounding.Interval.roundValue(key, interval);
-            // after rounding, since its day level (and above), its actually UTC!
-            // now apply post Tz
-            time = time + postTz.getOffset(time);
-            return time;
+            long localTime = Rounding.Interval.roundValue(key, interval);
+            if (!isUtc) {
+                return timeZone.convertLocalToUTC(localTime, true);
```

Comment:
Same question as above, only for TimeIntervalRounding I left the conversion here. Either way, in the end I'd like the two implementations to be similar, will change one or the other.